src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -354,5 +355,121 @@ public void ValidateGraphQLReservedNaming_DatabaseColumns(string dbColumnName, s
+        /// 2. For relationships not backed by an FK, there would be two referencing entities. This is because
+        /// at startup, We ÇANNOT determine which entity is the referencing entity and hence we keep ourselves open to the possibility
+        /// of either entity acting as the referencing entity. The actual referencing entity is determined during request execution.
+        /// </summary>
+        /// <param name="sourceEntityName">Source entity name.</param>
+        /// <param name="targetEntityName">Target entity name.</param>
+        /// <param name="expectedReferencingEntityNames">List of expected referencing entity names.</param>
+        private static void ValidateReferencingEntitiesForRelationship(
+            string sourceEntityName,
+            string targetEntityName,
+            List<string> expectedReferencingEntityNames)
+        {
+            _sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject sourceDbo);
+            _sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(targetEntityName, out DatabaseObject targetDbo);
+            DatabaseTable sourceTable = (DatabaseTable)sourceDbo;
+            DatabaseTable targetTable = (DatabaseTable)targetDbo;
+            List<ForeignKeyDefinition> foreignKeys = sourceDbo.SourceDefinition.SourceEntityRelationshipMap[sourceEntityName].TargetEntityToFkDefinitionMap[targetEntityName];
+            HashSet<DatabaseTable> expectedReferencingTables = new();
+            HashSet<DatabaseTable> actualReferencingTables = new();
+            foreach (string referencingEntityName in expectedReferencingEntityNames)
+            {
+                DatabaseTable referencingTable = referencingEntityName.Equals(sourceEntityName) ? sourceTable : targetTable;
+                expectedReferencingTables.Add(referencingTable);
+            }
+
+            foreach (ForeignKeyDefinition foreignKey in foreignKeys)
+            {
+                if (foreignKey.ReferencedColumns.Count == 0)
+                {
+                    continue;
+                }
+
+                DatabaseTable actualReferencingTable = foreignKey.Pair.ReferencingDbTable;
+                actualReferencingTables.Add(actualReferencingTable);
+            }
+
+            Assert.IsTrue(actualReferencingTables.SetEquals(expectedReferencingTables));
+        }
+
+        /// <summary>
+        /// Resets the database state and infers metadata for all the entities exposed in the config.
+        /// </summary>
+        private static async Task SetupTestFixtureAndInferMetadata()
+        {
+            TestHelper.SetupDatabaseEnvironment(DatabaseEngine);
+            RuntimeConfig runtimeConfig = SqlTestHelper.SetupRuntimeConfig();
+            RuntimeConfigProvider runtimeConfigProvider = TestHelper.GenerateInMemoryRuntimeConfigProvider(runtimeConfig);
+            SetUpSQLMetadataProvider(runtimeConfigProvider);
+            await ResetDbStateAsync();
```

Comment:
nit: suggestion to add a short comment about what `ResetDbStateAsync()` performs to avoid ambiguity on whether to invoke it before or after `_sqlMetadaProvider.InitializeAsync()` which is the method under test. 
```suggestion
           // The `ResetDbStateAsync()` method executes the .sql script of the respective database type. 
           // This serves as a setup phase for this test.   
            await ResetDbStateAsync();
```

Reply:
Added to method summary.